File: test/models/prospect_test.rb
require 'test_helper'

class ProspectTest < ActiveSupport::TestCase
  def setup
    @prospect = Prospect.new(email: "test@example.com", profile_link: "https://www.linkedin.com/in/linustorvalds", 
                             dev_link: "https://github.com/torvalds")
  end

  test "should accept valid submission" do
    assert @prospect.valid?
  end

  test "email should be present" do
    @prospect.email = "   "
    assert_not @prospect.valid?
  end

  test "email should not be too long" do
    @prospect.email = "denis" * 50 + "@example.com"
    assert_not @prospect.valid?
  end

  test "email validation should accept valid addresses" do
    valid_addresses = %w[user@example.com USER@foo.COM A_US-ER@foo.bar.org
                                 first.last@foo.jp alice+bob@baz.cn]
    valid_addresses.each do |valid_address|
      @prospect.email = valid_address
      assert @prospect.valid?, "#{valid_address.inspect} should be valid"
    end
  end

  test "email validation should reject invalid addresses" do
    invalid_addresses = %w[user@example,com user_at_foo.org user.name@example.
                                   foo@bar_baz.com foo@bar+baz.com]
    invalid_addresses.each do |invalid_address|
      @prospect.email = invalid_address
      assert_not @prospect.valid?, "#{invalid_address.inspect} should be invalid"
    end
  end
end

File: test/models/prospect_test.exs
defmodule Honeyworth.ProspectTest do
  use Honeyworth.ModelCase

  alias Honeyworth.Prospect

  @valid_attrs %{dev_link: "some content", email: "some content", name: "some content", profile_link: "some content"}
  @invalid_attrs %{}

  test "changeset with valid attributes" do
    changeset = Prospect.changeset(%Prospect{}, @valid_attrs)
    assert changeset.valid?
  end

  test "changeset without name attribute" do
    changeset = Prospect.changeset(%Prospect{}, Dict.delete(@valid_attrs, :name))
    assert changeset.valid?
  end

  test "changeset with invalid attributes" do
    changeset = Prospect.changeset(%Prospect{}, @invalid_attrs)
    refute changeset.valid?
  end
end
